### PR TITLE
OHM-204 Issue #576: Added mediators to heartbeat

### LINKED ISSUE #576: Add mediator uptime to heartbeat endpoint
Add the mediator uptime to https://github.com/jembi/openhim-core-js/blob/master/src/koaApi.coffee#L34

### PR DESCRIPTION
**Pull request by devcritter**
Closes #576 

@rcrichton if you could please review? thanks

### CODE REVIEW

File: src/api/heartbeat.coffee [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
+
+    mediators = yield Mediator.find().exec()
+    for mediator in mediators
+      if not result.mediators then result.mediators = {}
+
+      isUp = -> mediator._lastHeartbeat? and mediator._uptime? and
+      # have we received a heartbeat within the last minute?
+      moment().diff(mediator._lastHeartbeat, 'seconds') <= 60
+
+      if isUp mediator
```
Comment (rcrichton) [inline src/api/heartbeat.coffee]: It seems `isUp` doesn't take a `mediator` parameter, it just closes over the `mediator` variable in the current scope.

File: test/integration/heartbeatAPITest.coffee [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    afterEach (done) -> Mediator.remove {}, done
+
+    registerMediator = (done) ->
+      request("https://localhost:8080")
+        .post("/mediators")
+        .set("auth-username", testUtils.rootUser.email)
+        .set("auth-ts", authDetails.authTS)
+        .set("auth-salt", authDetails.authSalt)
+        .set("auth-token", authDetails.authToken)
```
Comment (rcrichton) [inline test/integration/heartbeatAPITest.coffee]: I think the heartbeats API is public, so I don't think the auth stuff is needed.

File: docs/dev-guide/api-ref.md [hunk 3]
```diff
@@ -635,3 +635,11 @@ For example a sample request could look like this:
...
 https://localhost:8080/logs?from=2015-10-28T12:31:46&until=2015-10-28T12:38:55&limit=100&start=10&level=error`
 ```
+
+#### Server uptime
+
+`GET heartbeat`
+
+returns 200 ok with `{ master: <core-uptime>, mediators: { <urn>: <mediator-uptime> ... }`
+
+Returns the server uptime in seconds. Includes a list of all registered mediators and if heartbeats have been received for them, will include their uptimes as well.
```
Comment (rcrichton) [inline docs/dev-guide/api-ref.md]: Perhaps mention that this is a public endpoint so not auth is required.

**Comment by rcrichton** [comment]
This looks good, just 2 small comments. :+1:

File: src/api/heartbeat.coffee (hunk 1 above)
Comment (devcritter) [inline src/api/heartbeat.coffee]: Aah oops, good catch

**Pull request merged**